internal/cluster/validator.go

```diff
@@ -101,11 +101,13 @@ func (v *clusterValidator) printIsMachineCidrDefined(context *clusterPreprocessC
 	case ValidationFailure:
 		if swag.BoolValue(context.cluster.VipDhcpAllocation) {
 			return "The Machine Network CIDR is undefined; setting the Machine Network CIDR initiates the VIPs DHCP lease allocation."
+		} else if common.IsSingleNodeCluster(context.cluster) {
+			return "The Machine Network CIDR is undefined; Setting Machine Network CIDR is required for single node cluster"
 		} else {
 			return "The Machine Network CIDR is undefined; the Machine Network CIDR can be defined by setting either the API or Ingress virtual IPs."
 		}
 	case ValidationSuccess:
-		if swag.BoolValue(context.cluster.UserManagedNetworking) {
+		if swag.BoolValue(context.cluster.UserManagedNetworking) && !common.IsSingleNodeCluster(context.cluster) {
```

Comment:
I think for IPv6, the machine CIDR must be set

Reply:
not changing logic for ipv6, just for single node. For single node in any case it must be set